File: CS/DXSample.Module/CriteriaReportParametersObject.cs
using System;
using System.ComponentModel;

using DevExpress.Xpo;

using DevExpress.ExpressApp;
using DevExpress.Persistent.Base;
using DevExpress.Persistent.BaseImpl;
using DevExpress.Persistent.Validation;
using DevExpress.ExpressApp.Reports;
using DevExpress.ExpressApp.Editors;
using DevExpress.Data.Filtering;

namespace DXSample.Module {
    [NonPersistent]
    public class CriteriaReportParametersObject : ReportParametersObjectBase {
        public CriteriaReportParametersObject(Session session) : base(session) { }

        public override DevExpress.Data.Filtering.CriteriaOperator GetCriteria() {
            UnitOfWork uow = Session as UnitOfWork;
            if (uow == null) {
                uow = new UnitOfWork(Session.DataLayer);
            }
            return CriteriaEditorHelper.GetCriteriaOperator(Criteria, ReportDataType, new ObjectSpace(uow, XafTypesInfo.Instance));
        }

        public override SortingCollection GetSorting() {
            return new SortingCollection();
        }

        private string _Criteria;
        [CriteriaObjectTypeMember("ReportDataType")]
        public string Criteria {
            get { return _Criteria; }
            set { SetPropertyValue("Criteria", ref _Criteria, value); }
        }
    }

}

File: VB/DXSample.Module/CriteriaReportParametersObject.vb
Imports System
Imports System.ComponentModel
Imports DevExpress.Xpo
Imports DevExpress.ExpressApp
Imports DevExpress.ExpressApp.Reports
Imports DevExpress.ExpressApp.Editors
Imports DevExpress.Data.Filtering

Namespace DXSample.Module

    <NonPersistent>
    Public Class CriteriaReportParametersObject
        Inherits ReportParametersObjectBase

        Public Sub New(ByVal objectSpace As IObjectSpace, ByVal reportDataType As Type)
            MyBase.New(objectSpace, reportDataType)
        End Sub

        Public Overrides Function GetCriteria() As CriteriaOperator
            Return CriteriaEditorHelper.GetCriteriaOperator(Criteria, ReportDataType, ObjectSpace)
        End Function

        Public Overrides Function GetSorting() As SortProperty()
            Return New SortProperty(-1) {}
        End Function

        Private _Criteria As String

        <CriteriaOptions("ReportDataType"), Size(SizeAttribute.Unlimited)>
        Public Property Criteria As String
            Get
                Return _Criteria
            End Get

            Set(ByVal value As String)
                _Criteria = value
            End Set
        End Property
    End Class
End Namespace
